clip_text.py

```diff
@@ -0,0 +1,55 @@
+from jina import DocumentArray, Executor, requests
+import torch
+import clip
+from typing import Iterable, Optional, List
+
+
+def _batch_generator(data: DocumentArray, batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i: i + batch_size]
+
+
+class CLIPTextEncoder(Executor):
+    """Encode text into embeddings."""
+
+    def __init__(self,
+                 model_name: str = 'ViT-B/32',
+                 default_batch_size: int = 32,
+                 default_traversal_paths: List[str] = ['r'],
+                 default_device: Optional[str] = 'cpu',
+                 jit: bool = True,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.device = default_device
+        self.model, _ = clip.load(model_name, 'cpu', jit)
```

Comment:
if we expose device we should use it to load rhe model.

Please take ClipImage as template

Reply:
Done